packages/core/src/__tests__/utils.test.ts

```diff
@@ -0,0 +1,31 @@
+import { isObjectLiteral } from '@/utils';
+
+describe('isObjectLiteral', () => {
+	test.each([
+		['empty object literal', {}, true],
+		['object with properties', { foo: 'bar', num: 123 }, true],
+		['nested object literal', { nested: { foo: 'bar' } }, true],
+		['null', null, false],
+		['empty array', [], false],
+		['array with values', [1, 2, 3], false],
+		['number', 42, false],
+		['string', 'string', false],
+		['boolean', true, false],
+		['undefined', undefined, false],
+		['Date object', new Date(), false],
+		['RegExp object', new RegExp(''), false],
+		['Map object', new Map(), false],
+		['Set object', new Set(), false],
+		['arrow function', () => {}, false],
+		['regular function', function () {}, false],
+		['class instance', new (class TestClass {})(), false],
+		['object with custom prototype', Object.create({ customMethod: () => {} }), false],
+		['Object.create(null)', Object.create(null), false],
+	])('should return %s for %s', (_, input, expected) => {
+		expect(isObjectLiteral(input)).toBe(expected);
+	});
```

Comment:
Some more to consider:
- does an object with a symbol key pass?
- does a `Proxy` pass?
- does a `Promise` pass?

Reply:
> does an object with a symbol key pass?

Yes

> does a `Proxy` pass?

We can't reliable detect proxies

> does a `Promise` pass?

Nop. I'll add a test.